development/build/utils.js

```diff
@@ -218,8 +228,67 @@ function getPathInsideNodeModules(packageName, pathToFiles) {
   return targetPath;
 }
 
+/**
+ * Get the name for the current build.
+ *
+ * @param {object} options - The build options.
+ * @param {string} options.buildType - The build type of the current build.
+ * @param {boolean} options.applyLavaMoat - Flag if lavamoat was applied.
+ * @param {boolean} options.shouldIncludeSnow - Flag if snow should be included in the build name.
+ * @param options.environment
+ * @returns {string} The build name.
+ */
+function getBuildName({
+  environment,
+  buildType,
+  applyLavaMoat,
+  shouldIncludeSnow,
+}) {
+  if (environment === ENVIRONMENT.PRODUCTION) {
+    return 'MetaMask';
+  }
+
+  const mv3Str = process.env.ENABLE_MV3 ? ' MV3' : '';
+  const lavamoatStr = applyLavaMoat ? ' lavamoat' : '';
+  const snowStr = shouldIncludeSnow ? ' snow' : '';
+
+  return buildType === 'mmi'
+    ? `MetaMask Institutional ${mv3Str}`
+    : `MetaMask ${capitalize(buildType)}${mv3Str}${lavamoatStr}${snowStr}`;
```

Comment:
TODO: figure out what's wrong with this logic or the callers of this helper func. For some reason, `snow` ends up in the build name used to announce the provider for 6963.

Reply:
Were these suffixes supposed to be removed per the comment [here](https://github.com/MetaMask/metamask-extension/pull/19908#issuecomment-1739806953)?

Reply:
correct. This helper func is also used for generating the extension name in the manifest file so these need to stay here.

https://github.com/MetaMask/metamask-extension/pull/19908/files#diff-a05d1f077953d24c0d8ddecf64163da2653ba3b728e1ba1f18af6d770767b127R1187

https://github.com/MetaMask/metamask-extension/pull/19908/files#diff-23d7dfe51f90119effef1172a76156b26bb03631109f2d317b13c4ccce4eefc4R134